overlays/emacs.nix

```diff
@@ -73,17 +73,31 @@ let
         drv: drv.overrideAttrs (old:
           let
             libName = drv: super.lib.removeSuffix "-grammar" drv.pname;
-            lib = drv: ''lib${libName drv}.so'';
-            linkCmd = drv: "ln -s ${drv}/parser $out/lib/${lib drv}";
+            libSuffix = if super.stdenv.isDarwin then "dylib" else "so";
+            lib = drv: ''lib${libName drv}.${libSuffix}'';
+            linkCmd = drv:
+              if super.stdenv.isDarwin
+              then ''cp ${drv}/parser $out/lib/${lib drv}
+                     /usr/bin/install_name_tool -id $out/lib/${lib drv} $out/lib/${lib drv}
+                     /usr/bin/codesign -s - -f $out/lib/${lib drv}
+                ''
+              else ''ln -s ${drv}/parser $out/lib/${lib drv}'';
             linkerFlag = drv: "-l" + libName drv;
+
```

Comment:
Adding the newline here makes `git blame` a bit more awkward, so I'd prefer to leave it unchanged.